src/lib/Microsoft.Health.Events/EventHubProcessor/ResumableAssignedPartitionProcessor.cs

```diff
@@ -0,0 +1,51 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Events.EventHubProcessor
+{
+    public class ResumableAssignedPartitionProcessor : IResumableEventProcessor
+    {
+        private PartitionLockingBackgroundService _lockingBackgroundService;
+
+        private bool _isRunning = false;
+
+        public ResumableAssignedPartitionProcessor(PartitionLockingBackgroundService lockingBackgroundService)
+        {
+            _lockingBackgroundService = lockingBackgroundService;
+        }
+
+        public void Dispose()
+        {
+            // todo?
+        }
+
+        public async Task ResumeAsync(CancellationToken ct)
+        {
+            if (!_isRunning)
```

Comment:
Instead of a bool you may want to use an int to represent `_isRunning`, and use `Interlocked.Exchange()` to set this value. Right now this implementation is not thread safe, and there is a slight possibility that `_isRunning` can become out of sync with the actual state.

```c
public async Task ResumeAsync(CancellationToken ct)
{
    if (Interlocked.Exchange(ref _isRunning, 1) == 0)
    {
        await _lockingBackgroundService.StartAsync(ct);
    }
}

public async Task RunAsync(CancellationToken ct)
{
    if (Interlocked.Exchange(ref _isRunning, 1) == 0)
    {
        await _lockingBackgroundService.StartAsync(ct);
    }
}

public async Task SuspendAsync(CancellationToken ct)
{
    if (Interlocked.Exchange(ref _isRunning, 0) == 1)
    {
        await _lockingBackgroundService.StopAsync(ct);
    }
}
```